### PR TITLE
Adding customer admin sync test.

### PR DESCRIPTION
**Pull request by kwoodson**
This pull request adds a test for aad sync.

Relies on https://github.com/openshift/release/pull/3022

Fixes https://github.com/openshift/openshift-azure/issues/1222

```release-note
None
```

### CODE REVIEW

**Comment by kwoodson** [comment]
I expect test failures until the release PR is merged.

File: test/e2e/specs/customer_admin.go [hunk 1]
```diff
@@ -218,5 +219,26 @@ var _ = Describe("Openshift on Azure customer-admin e2e tests [CustomerAdmin][Fa
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	It("should sync AAD admin group", func() {
+		syncSecret, err := cli.Client.Admin.CoreV1.Secrets("openshift-infra").Get("aad-group-sync-config", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		idPre := "44e69b"
+		idSuf := "3a8907"
```
Comment (asalkeld) [inline test/e2e/specs/customer_admin.go]: are you sure this is correct? the objectID ends in "730d7b0" when i go look at it

Comment (kwoodson) [reply test/e2e/specs/customer_admin.go]: @asalkeld, thanks.  Will update.

**Comment by kwoodson** [comment]
```
Openshift on Azure customer-admin e2e tests [CustomerAdmin][Fake] 
  should sync AAD admin group
  /home/kwoodson/go/src/github.com/openshift/openshift-azure/test/e2e/specs/customer_admin.go:244
•SSSS
Ran 8 of 26 Specs in 31.732 seconds
SUCCESS! -- 8 Passed | 0 Failed | 0 Pending | 18 Skipped
--- PASS: TestE2E (31.73s)
PASS
ok  	github.com/openshift/openshift-azure/test/e2e	31.747s
signal: terminated
```

File: test/e2e/specs/customer_admin.go [hunk 2]
```diff
@@ -240,5 +241,26 @@ var _ = Describe("Openshift on Azure customer-admin e2e tests [CustomerAdmin][Fa
...
+			oca, err := cli.Client.Admin.UserV1.Groups().Get("osa-customer-admins", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			found := false
+			for _, user := range oca.Users {
+				if strings.HasPrefix(user, testUser) {
+					found = true
+				}
+			}
+			Expect(found).To(BeTrue())
```
Comment (jim-minter) [inline test/e2e/specs/customer_admin.go]: should test that there is one and only one user, and that it matches

File: test/e2e/specs/customer_admin.go [hunk 3]
```diff
@@ -240,5 +241,26 @@ var _ = Describe("Openshift on Azure customer-admin e2e tests [CustomerAdmin][Fa
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	It("should sync AAD admin group", func() {
+		syncSecret, err := cli.Client.Admin.CoreV1.Secrets("openshift-infra").Get("aad-group-sync-config", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		idPre := "44e69b"
```
Comment (jim-minter) [inline test/e2e/specs/customer_admin.go]: I don't believe it's an excessive risk to check the full guid or full username in.

File: test/e2e/specs/customer_admin.go (hunk 2 above)
Comment (kwoodson) [reply test/e2e/specs/customer_admin.go]: I will update this.

File: test/e2e/specs/customer_admin.go (hunk 3 above)
Comment (kwoodson) [reply test/e2e/specs/customer_admin.go]: I agree that the guid is probably fine.  I'd still prefer not to expose the account information.

**Comment by kwoodson** [comment]
@asalkeld @jim-minter 
I have updated the PR, please review.

**Comment by kwoodson** [comment]
@mjudeikis This upgrade failure is obviously a flake.  This PR is only adding a test case.

```
2019/03/07 16:04:56 Executing test e2e-upgrade-v2.5
{"component":"entrypoint","level":"error","msg":"Process did not finish before 4h0m0s timeout","time":"2019-03-07T20:03:49Z"}
2019/03/07 20:03:49 error: Process interrupted with signal interrupt, exiting in 2s ...
2019/03/07 20:03:49 cleanup: Deleting test pod e2e-upgrade-v2.5
{"component":"entrypoint","level":"error","msg":"Process gracefully exited before 15s grace period","time":"2019-03-07T20:03:51Z"}
```

**Pull request merged**